rust/main/hyperlane-metric/src/utils.rs

```diff
@@ -5,13 +5,18 @@ use url::Url;
 /// converts url into a host:port string
 pub fn url_to_host_info(url: &Url) -> Option<String> {
     let mut s = String::new();
+
     if let Some(host) = url.host_str() {
         s.push_str(host);
-        if let Some(port) = url.port() {
+        if let Some(port) = url.port_or_known_default() {
             write!(&mut s, ":{port}").unwrap();
         }
         Some(s)
     } else {
-        None
+        let url_str = url.to_string();
+        match url_str.split_once('/') {
+            Some((left, _)) => Some(left.into()),
+            None => Some(url_str),
+        }
```

Comment:
This assumes the only reason parsing could fail is a missing [scheme](https://docs.rs/url/latest/url/struct.Url.html#method.scheme). Wdyt of appending a dummy scheme if one is missing and still parsing using `host_str`, since that's more battle tested.
```suggestion
// `url` has to be `mut`
if url.scheme().is_empty() {
            let with_dummy_scheme = format!("http://{}", input);
            url = Url::parse(&with_scheme)?;
}

    if let Some(host) = url.host_str() {
        s.push_str(host);
        if let Some(port) = url.port_or_known_default() {
            write!(&mut s, ":{port}").unwrap();
        }
        Some(s)
    } else {
        None


```

Reply:
yeah, that works too, let me update it